base_dav/models/dav_collection.py

```diff
@@ -0,0 +1,300 @@
+        for this in self:
+            this.url = '%s%s/%s/%s' % (
+                base_url,
+                PREFIX,
+                self.env.user.login,
+                this.id,
+            )
+
+    @api.constrains('domain')
+    def _check_domain(self):
+        self._eval_domain()
+
+    @api.model
+    def _eval_context(self):
+        return {
+            'user': self.env.user,
+        }
+
+    @api.multi
+    def _eval_domain(self):
+        self.ensure_one()
+        return list(tools.safe_eval(self.domain, self._eval_context()))
+
+    @api.multi
+    def eval(self):
+        if not self:
+            return self.env['unknown']
+        self.ensure_one()
+        return self.env[self.model_id.model].search(self._eval_domain())
+
+    @api.multi
+    def get_record(self, components):
+        self.ensure_one()
+        collection_model = self.env[self.model_id.model]
+
+        field_name = self.field_uuid.name or "id"
+        domain = [(field_name, '=', components[-1])] + self._eval_domain()
+        return collection_model.search(domain, limit=1)
+
+    @api.multi
+    def from_vobject(self, item):
+        self.ensure_one()
+
+        result = {}
+        if self.dav_type == 'calendar':
+            if item.name != 'VCALENDAR':
+                return None
+            if not hasattr(item, 'vevent'):
+                return None
+            item = item.vevent
+        elif self.dav_type == 'addressbook' and item.name != 'VCARD':
+            return None
+
+        children = {c.name.lower(): c for c in item.getChildren()}
+        for mapping in self.field_mapping_ids:
+            name = mapping.name.lower()
+            if name not in children:
+                continue
+
+            if name in children:
+                value = mapping.from_vobject(children[name])
+                if value:
+                    result[mapping.field_id.name] = value
+
+        return result
+
+    @api.multi
+    def to_vobject(self, record):
+        self.ensure_one()
+        result = None
+        vobj = None
+        if self.dav_type == 'calendar':
+            result = vobject.iCalendar()
+            vobj = result.add('vevent')
+        if self.dav_type == 'addressbook':
+            result = vobject.vCard()
+            vobj = result
+        for mapping in self.field_mapping_ids:
+            value = mapping.to_vobject(record)
+            if value:
+                vobj.add(mapping.name).value = value
+
+        if 'uid' not in vobj.contents:
+            vobj.add('uid').value = '%s,%s' % (record._name, record.id)
+        if 'rev' not in vobj.contents and 'write_date' in record._fields:
+            vobj.add('rev').value = record.write_date.strftime('%Y-%m-%dT%H%M%SZ')
+        return result
+
+    @api.model
+    def _odoo_to_http_datetime(self, value):
+        return time.strftime(
+            '%a, %d %b %Y %H:%M:%S GMT',
+            value.timetuple(),
+        )
```

Comment:
> you'll need to increase coverage for this to be mergeable.
> 
> I'd also suggest to update to the 3.x branch of radicale: https://radicale.org/v3.html
Radicale v3 needs Python 3.6. I have had to change Python version in Travis CI because in repo it's 3.5. I don't know if it is a good idea...